CRM/Core/BAO/CustomGroup.php

```diff
@@ -2349,4 +2276,103 @@ public static function getCustomGroupExtendsOptions() {
     return $options;
   }
 
+  /**
+   * Get custom data for display on CiviCRM pages.
+   *
+   * @param string $dataType
+   * @param mixed $fieldData
+   * @param int|null $entityIDFieldValue
+   * @param int|null $id
+   *   ID of the record in the relevant custom data table.
+   *   Oddly this can be NULL - it rather feels like we should not
+   *   call this function if so - perhaps we can iterate on that.
+   * @param int $fieldID
+   * @param string $table
+   *
+   * @return array
+   * @throws \CRM_Core_Exception
+   * @internal do not call this from outside core code - the signature is expected to change multiple times.
+   *
+   */
+  public static function getCustomDisplayValue(string $dataType, $fieldData, ?int $entityIDFieldValue, ?int $id, int $fieldID, string $table): array {
```

Comment:
@eileenmcnaughton Are you intending to access this from other classes? Otherwise we could keep it private for now?

Reply:
@mattwire I was when I did this - but wound up leaving it out of scope in https://github.com/civicrm/civicrm-core/pull/30981 because once I worked through it the only thing this function does is make the file custom fields display correctly  - & it turns out to be broken before & after that PR - so perhaps it does make sense to make this private for now - depending how quickly things move on that PR I might fix the file custom fields but maybe when I get to it is the time to